src/validate.rs

```diff
@@ -900,6 +890,33 @@ fn validate_repos(data: &Data, errors: &mut Vec<String>) {
     });
 }
 
+fn check_team_access(
+    data: &Data,
+    all_github_teams: &HashSet<(String, String)>,
+    repo: &Repo,
+    team_name: &str,
+) -> anyhow::Result<()> {
+    let team = data
+        .team(team_name)
+        .with_context(|| format!("The file 'teams/{team_name}.toml' does not exist."))?;
+    let github_teams = team.github_teams(data)?;
+    let mut team_found = false;
+    for github_team in &github_teams {
+        if all_github_teams.contains(&(repo.org.clone(), github_team.name.to_string())) {
```

Comment:
Why do you check `all_github_teams`? I don't think that we expect the corresponding GH team to come from anywhere else than the `team` itself. So essentially we want something like this:
```rust
let gh_teams: Vec<_> = github_teams.iter().filter(|t| t.org == repo.org).collect();
if gh_teams.is_empty() {
    bail!(
        "The team '{team_name}' does not have any GitHub teams in the org '{}'.",
        repo.org
    );
}
```